frontend/amp/components/elements/RichLinkBlockComponent.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { css } from 'emotion';
+import { palette } from '@guardian/pasteup/palette';
+import { serif, sans } from '@guardian/pasteup/fonts';
+import { pillarPalette } from '@frontend/lib/pillars';
+
+export const RichLinkBlockComponent: React.SFC<{
+    element: RichLinkBlockElement;
+    pillar: Pillar;
+}> = ({ element, pillar }) => {
+    if (element.sponsorship) {
+        throw new Error('Sponsored rich links not supported');
+    }
+    return (
+        <aside className={style}>
+            <a className={linkStyle(pillar)} href={element.url}>
+                {element.text}
+            </a>
+        </aside>
+    );
+};
+
+const style = css`
```

Comment:
Interesting that we declare the component *above* the associated styles here. I'm not sure what I think about this (my slight inclination is that having consts in use order is helpful but I haven't thought about it much) - but it would be nice to be consistent here in any case cc @SiAdcock ?